### PR TITLE
[DX-218] Expose cli to be used programmatically

### LINKED ISSUE #441: Running local instance of registry via node does not create _package folders
**Background** : We are working on enabling developers to be able to write integration test suites for their components, to do this one of the things needed is to be able to bring up and stop a local instance of the registry.

**Code** : 
```
startRegistry: function(done) {
            registry = new oc.Registry({
                baseUrl: 'http://localhost:3030/',
                local: true,
                port: 3030,
                hotReloading: false,
                path: path.resolve('../'),
                env: {name: 'local'},
                verbosity: 1,
                discovery: true
            });
            registry.start(done);
        }
```
Pretty straight forward snippet. Note that we need discovery true so that we can use phantomJS or nightmare js or casper ... to be able to load the component in preview mode.

**Problem** : So the registry did start up but it does not generate the _package folder for the component, which means we cannot really load the component in preview mode.

Thoughts?

### PR DESCRIPTION
**Pull request by nickbalestra**
Fixes #441 by exposing the dev interface and allowing acceptance testing in a cleaner way

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+  const dependencies = {
+    local: new Local(),
+    logger: options.logger || { log() {} }
+  };
+  const opts = _.omit(options, 'logger');
+
+  require(`./cli/facade/${cmd}.js`)(dependencies)(opts, cb);
+}
+
+module.exports.cli = cliCommand;
```
Comment (matteofigus) [inline src/index.js]: can we move lines 3-14 to own file inside the cli folder?

File: src/index.js [hunk 2]
```diff
@@ -1,4 +1,18 @@
 'use strict';
 
+const _ = require('lodash');
+const Local = require('./cli/domain/local');
+
+function cliCommand(cmd, options, cb) {
+  const dependencies = {
+    local: new Local(),
+    logger: options.logger || { log() {} }
```
Comment (matteofigus) [inline src/index.js]: the logger needs more methods: https://github.com/opentable/oc/blob/master/src/cli/logger.js#L14-L17

File: src/index.js [hunk 3]
```diff
@@ -1,4 +1,8 @@
 'use strict';
 
+const _ = require('lodash');
+const Local = require('./cli/domain/local');
```
Comment (matteofigus) [inline src/index.js]: forgot about those 2 ^ ? Everything else LGTM

**Comment by nickbalestra** [comment]
@matteofigus 
Fixed, there also were some wrong paths, and I've also added some unit tests. Should be good to review now

**Pull request merged**